### PR TITLE
APS-4451 - Fix issues with SharedMetricsRegistry and MDCLloggingEC wa…

### PR DESCRIPTION
**Pull request by AndySpaven**
…rnings

### CODE REVIEW

File: app/controllers/Controller.scala [hunk 1]
```diff
@@ -120,11 +121,22 @@ abstract class LoggedOutController()
...
+}
+
 abstract class BaseController()
-  extends AuthConfigImpl with I18nSupport with FrontendController with HeaderEnricher {
+  extends AuthConfigImpl with I18nSupport with HeaderCarrierConversion with HeaderEnricher {
 
   val errorHandler: ErrorHandler
   val sessionService: SessionService
+
+  implicit def ec: ExecutionContext
```
Comment (peteslater-ee) [inline app/controllers/Controller.scala]: Any reason this is not a val?

Comment (AndySpaven) [reply app/controllers/Controller.scala]: Because it's abstract.

Comment (peteslater-ee) [reply app/controllers/Controller.scala]: Doesn't need to be a def though

Comment (AndySpaven) [reply app/controllers/Controller.scala]: It should be defined as such when abstract in a trait. This allows someone to override it any way they want (including with lazy).  A val does not.

**Pull request merged**